x-pack/plugins/reporting/server/lib/store/store.test.ts

```diff
@@ -395,12 +409,57 @@ describe('ReportingStore', () => {
             },
           },
           "id": "vastly-great-report-id",
-          "if_primary_term": undefined,
-          "if_seq_no": undefined,
+          "if_primary_term": 10002,
+          "if_seq_no": 42,
           "index": ".reporting-test-index-12345",
           "refresh": true,
         },
       ]
     `);
   });
+
+  it('prepareReportForRetry resets the expiration and status on the report document', async () => {
+    const store = new ReportingStore(mockCore, mockLogger);
+    const report = new Report({
+      _id: 'pretty-good-report-id',
+      _index: '.reporting-test-index-94058763',
+      _seq_no: 42,
+      _primary_term: 10002,
+      jobtype: 'test-report-2',
+      created_by: 'created_by_test_string',
+      browser_type: 'browser_type_test_string',
+      status: 'processing',
+      process_expiration: '2002',
+      max_attempts: 3,
+      payload: {
+        title: 'test report',
+        headers: 'rp_test_headers',
+        objectType: 'testOt',
+        browserTimezone: 'utc',
+      },
+      timeout: 30000,
+    });
+
+    await store.prepareReportForRetry(report);
+
+    const [updateCall] = mockEsClient.update.mock.calls;
+    expect(updateCall).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "body": Object {
+            "doc": Object {
+              "migration_version": "7.14.0",
+              "process_expiration": null,
+              "status": "pending",
```

Comment:
Maybe it would be clearer to use here instead of inline mock assert a couple of specific assertions which assert on what is mentioned in test title.